returnn/tf/layers/rec.py

```diff
@@ -985,6 +985,8 @@ def __init__(self, net_dict, source_data, time_dim_tag, rec_layer_name, parent_n
     self.parent_rec_layer = None  # type: typing.Optional[RecLayer]
     self.parent_net = parent_net
     self.net_dict = safe_deep_copy(net_dict)
+    self.rec_layer_name = rec_layer_name
+    self.parent_get_layer = parent_get_layer
```

Comment:
I think copying `parent_get_layer` is a bad idea (can potentially lead to very nasty problems). You only need that for your copy of `_SubnetworkRecCell`? Then this can be removed after the other cleanup I suggested before.